Sources/ProjectDescription/FileList.swift

```diff
@@ -2,28 +2,54 @@ import Foundation
 
 public struct FileList: Codable, Equatable {
     /// List glob patterns.
-    public let globs: [Path]
+    public let globs: [FileListGlob]
 
     /// Initializes the files list with the glob patterns.
     ///
-    /// - Parameter globs: Glob patterns.
-    public init(globs: [Path]) {
+    ///   - glob: Relative glob pattern.
+    ///   - excluding: Relative glob patterns for excluded files.
+    public static func list(_ globs: [FileListGlob]) -> FileList {
+        .init(globs)
+    }
+
+    /// Initializes the files list with the glob patterns.
+    ///
+    ///   - glob: Relative glob pattern.
+    ///   - excluding: Relative glob patterns for excluded files.
+    private init(_ globs: [FileListGlob]) {
         self.globs = globs
     }
 
-    public static func == (lhs: FileList, rhs: FileList) -> Bool {
-        lhs.globs == rhs.globs
+    // for backward compatibility when globs property had [Path] type
+    public init(globs: [Path]) {
+        self.init(globs.map { .glob($0) })
+    }
+
+    public enum CodingKeys: String, CodingKey {
+        case globs
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        // previously globs property had [Path] type
+        if let paths = try? container.decode([Path].self, forKey: .globs) {
+            self.init(globs: paths)
+        } else {
+            let globs = try container.decode([FileListGlob].self, forKey: .globs)
+            self.init(globs)
+        }
     }
```

Comment:
Why do we need this? Do we ever decode serialized `ProjectDescription.FileList`?

Reply:
declared as Codable, 
I don't know full flow of tuist under the hood - if there are any persistent store (cache?) it will be corrupted,
so I just wanna make safe code

Reply:
We shouldn't persist anything from Project description AFAIK, Right @pepicrft @fortmarek ?

Reply:
We don't persist it so we don't need this code.

Reply:
removed